rpcclient/wallet.go

```diff
@@ -2367,6 +2367,91 @@ func (c *Client) GetInfo() (*btcjson.InfoWalletResult, error) {
+// provided by the server.
+func (r FutureWalletCreateFundedPsbtResult) Receive() (*btcjson.WalletCreateFundedPsbtResult, error) {
+	res, err := receiveFuture(r)
+	if err != nil {
+		return nil, err
+	}
+
+	// Unmarshal result as a getinfo result object.
+	var psbtRes btcjson.WalletCreateFundedPsbtResult
+	err = json.Unmarshal(res, &psbtRes)
+	if err != nil {
+		return nil, err
+	}
+
+	return &psbtRes, nil
+}
+
+// WalletCreateFundedPsbtAsync returns an instance of a type that can be used
+// to get the result of the RPC at some future time by invoking the Receive
+// function on the returned instance.
+//
+// See WalletCreateFundedPsbt for the blocking version and more details.
+func (c *Client) WalletCreateFundedPsbtAsync(
+	inputs []btcjson.PsbtInput, outputs []btcjson.PsbtOutput, locktime *int64,
+	options *btcjson.WalletCreateFundedPsbtOpts, bip32Derivs *bool,
+) FutureWalletCreateFundedPsbtResult {
+	cmd := btcjson.NewWalletCreateFundedPsbtCmd(inputs, outputs, locktime, options, bip32Derivs)
+	return c.sendCmd(cmd)
+}
+
+// WalletCreateFundedPsbt creates and funds a transaction in the Partially
+// Signed Transaction format. Inputs will be added if supplied inputs are not
+// enough.
+func (c *Client) WalletCreateFundedPsbt(
+	inputs []btcjson.PsbtInput, outputs []btcjson.PsbtOutput, locktime *int64,
+	options *btcjson.WalletCreateFundedPsbtOpts, bip32Derivs *bool,
+) (*btcjson.WalletCreateFundedPsbtResult, error) {
+	return c.WalletCreateFundedPsbtAsync(inputs, outputs, locktime, options, bip32Derivs).Receive()
+}
+
+// FutureWalletProcessPsbtResult is a future promise to deliver the result of a
+// WalletCreateFundedPsb RPC invocation (or an applicable error).
+type FutureWalletProcessPsbtResult chan *response
+
+// Receive waits for the response promised by the future and returns the info
+// provided by the server.
+func (r FutureWalletProcessPsbtResult) Receive() (*btcjson.WalletProcessPsbtResult, error) {
+	res, err := receiveFuture(r)
+	if err != nil {
+		return nil, err
+	}
+
+	// Unmarshal result as a getinfo result object.
+	var psbtRes btcjson.WalletProcessPsbtResult
+	err = json.Unmarshal(res, &psbtRes)
+	if err != nil {
+		return nil, err
+	}
+
+	return &psbtRes, nil
+}
+
+// WalletProcessPsbtAsync returns an instance of a type that can be used
+// to get the result of the RPC at some future time by invoking the Receive
+// function on the returned instance.
+//
+// See WalletProcessPsbt for the blocking version and more details.
+func (c *Client) WalletProcessPsbtAsync(
+	psbt string, sign *bool, sighashType *string, bip32Derivs *bool,
+) FutureWalletProcessPsbtResult {
+	cmd := btcjson.NewWalletProcessPsbtCmd(psbt, sign, sighashType, bip32Derivs)
+	return c.sendCmd(cmd)
+}
+
+// WalletProcessPsbt updates a PSBT with input information from our wallet and
+// then signs inputs.
+func (c *Client) WalletProcessPsbt(
+	psbt string, sign *bool, sighashType *string, bip32Derivs *bool,
```

Comment:
Why not have an enum type for `sighashType`? The possible values are quite limited and clearly documented in Bitcoin Core.

```
"ALL"
"NONE"
"SINGLE"
"ALL|ANYONECANPAY"
"NONE|ANYONECANPAY"
"SINGLE|ANYONECANPAY"
```